src/Microsoft.AspNet.Security.OAuthBearer/OAuthBearerAuthenticationMiddleware.cs

```diff
@@ -0,0 +1,117 @@
+    {
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Bearer authentication component which is added to an HTTP pipeline. This constructor is not
+        /// called by application code directly, instead it is added by calling the the IAppBuilder UseOAuthBearerAuthentication 
+        /// extension method.
+        /// </summary>
+        public OAuthBearerAuthenticationMiddleware(
+            RequestDelegate next,
+            IServiceProvider services,
+            IDataProtectionProvider dataProtectionProvider,
+            ILoggerFactory loggerFactory,
+            IOptions<OAuthBearerAuthenticationOptions> options,
+            ConfigureOptions<OAuthBearerAuthenticationOptions> configureOptions)
+            : base(next, services, options, configureOptions)
+        {
+            _logger = loggerFactory.Create<OAuthBearerAuthenticationMiddleware>();
+            if (Options.Notifications == null)
+            {
+                Options.Notifications = new OAuthBearerAuthenticationNotifications();
+            }
+
+            if (Options.SecurityTokenValidators == null)
+            {
+                Options.SecurityTokenValidators = new List<ISecurityTokenValidator> { new JwtSecurityTokenHandler() };
+            }
+
+            if (string.IsNullOrWhiteSpace(Options.TokenValidationParameters.ValidAudience) && !string.IsNullOrWhiteSpace(Options.Audience))
+            {
+                Options.TokenValidationParameters.ValidAudience = Options.Audience;
+            }
+
+            if (Options.ConfigurationManager == null)
+            {
+                if (Options.Configuration != null)
+                {
+                    Options.ConfigurationManager = new StaticConfigurationManager<OpenIdConnectConfiguration>(Options.Configuration);
+                }
+                else if (!(string.IsNullOrWhiteSpace(Options.MetadataAddress) && string.IsNullOrWhiteSpace(Options.Authority)))
+                {
+                    if (string.IsNullOrWhiteSpace(Options.MetadataAddress) && !string.IsNullOrWhiteSpace(Options.Authority))
+                    {
+                        Options.MetadataAddress = Options.Authority;
+                        if (!Options.MetadataAddress.EndsWith("/", StringComparison.Ordinal))
+                        {
+                            Options.MetadataAddress += "/";
+                        }
+
+                        Options.MetadataAddress += ".well-known/openid-configuration";
+                    }
+
+                    HttpClient httpClient = new HttpClient(ResolveHttpMessageHandler(Options));
+                    httpClient.Timeout = Options.BackchannelTimeout;
+                    httpClient.MaxResponseContentBufferSize = 1024 * 1024 * 10; // 10 MB
+
+                    Options.ConfigurationManager = new ConfigurationManager<OpenIdConnectConfiguration>(Options.MetadataAddress, httpClient);
```

Comment:
This kind of use will definitely be a niche market (I'd bet that opaque tokens will at least be used in 95% of the real world scenarios), so it shouldn't be enabled by default (see my remark concerning the split of this middleware).

@brentschmaltz but more importantly, do you expect the access token to be signed using the same certificate or RSA key exposed on the JWKS endpoint and used to verify the `id_token`? For obvious reasons, both shouldn't be signed using the same certificate (or your bearer middleware will end up accepting `id_token`s as valid `access_token`s).

I'll ping you on JabbR.